commit ff49c86f27e4726a86f5034543e6e684daf41955
Merge: b97d4c424e36 75e91c888989
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Thu Dec 17 11:18:00 2020 -0800

    Merge tag 'f2fs-for-5.11-rc1' of git://git.kernel.org/pub/scm/linux/kernel/git/jaegeuk/f2fs
    
    Pull f2fs updates from Jaegeuk Kim:
     "In this round, we've made more work into per-file compression support.
    
      For example, F2FS_IOC_GET | SET_COMPRESS_OPTION provides a way to
      change the algorithm or cluster size per file. F2FS_IOC_COMPRESS |
      DECOMPRESS_FILE provides a way to compress and decompress the existing
      normal files manually.
    
      There is also a new mount option, compress_mode=fs|user, which can
      control who compresses the data.
    
      Chao also added a checksum feature with a mount option so that
      we are able to detect any corrupted cluster.
    
      In addition, Daniel contributed casefolding with encryption patch,
      which will be used for Android devices.
    
      Summary:
    
      Enhancements:
       - add ioctls and mount option to manage per-file compression feature
       - support casefolding with encryption
       - support checksum for compressed cluster
       - avoid IO starvation by replacing mutex with rwsem
       - add sysfs, max_io_bytes, to control max bio size
    
      Bug fixes:
       - fix use-after-free issue when compression and fsverity are enabled
       - fix consistency corruption during fault injection test
       - fix data offset for lseek
       - get rid of buffer_head which has 32bits limit in fiemap
       - fix some bugs in multi-partitions support
       - fix nat entry count calculation in shrinker
       - fix some stat information
    
      And, we've refactored some logics and fix minor bugs as well"
    
    * tag 'f2fs-for-5.11-rc1' of git://git.kernel.org/pub/scm/linux/kernel/git/jaegeuk/f2fs: (36 commits)
      f2fs: compress: fix compression chksum
      f2fs: fix shift-out-of-bounds in sanity_check_raw_super()
      f2fs: fix race of pending_pages in decompression
      f2fs: fix to account inline xattr correctly during recovery
      f2fs: inline: fix wrong inline inode stat
      f2fs: inline: correct comment in f2fs_recover_inline_data
      f2fs: don't check PAGE_SIZE again in sanity_check_raw_super()
      f2fs: convert to F2FS_*_INO macro
      f2fs: introduce max_io_bytes, a sysfs entry, to limit bio size
      f2fs: don't allow any writes on readonly mount
      f2fs: avoid race condition for shrinker count
      f2fs: add F2FS_IOC_DECOMPRESS_FILE and F2FS_IOC_COMPRESS_FILE
      f2fs: add compress_mode mount option
      f2fs: Remove unnecessary unlikely()
      f2fs: init dirty_secmap incorrectly
      f2fs: remove buffer_head which has 32bits limit
      f2fs: fix wrong block count instead of bytes
      f2fs: use new conversion functions between blks and bytes
      f2fs: rename logical_to_blk and blk_to_logical
      f2fs: fix kbytes written stat for multi-device case
      ...

diff --cc fs/f2fs/checkpoint.c
index 54a1905af052,617d0f6b0836..897edb7c951a
--- a/fs/f2fs/checkpoint.c
+++ b/fs/f2fs/checkpoint.c
@@@ -1385,6 -1385,27 +1385,26 @@@ static void commit_checkpoint(struct f2
  	f2fs_submit_merged_write(sbi, META_FLUSH);
  }
  
+ static inline u64 get_sectors_written(struct block_device *bdev)
+ {
 -	return bdev->bd_part ?
 -		(u64)part_stat_read(bdev->bd_part, sectors[STAT_WRITE]) : 0;
++	return (u64)part_stat_read(bdev, sectors[STAT_WRITE]);
+ }
+ 
+ u64 f2fs_get_sectors_written(struct f2fs_sb_info *sbi)
+ {
+ 	if (f2fs_is_multi_device(sbi)) {
+ 		u64 sectors = 0;
+ 		int i;
+ 
+ 		for (i = 0; i < sbi->s_ndevs; i++)
+ 			sectors += get_sectors_written(FDEV(i).bdev);
+ 
+ 		return sectors;
+ 	}
+ 
+ 	return get_sectors_written(sbi->sb->s_bdev);
+ }
+ 
  static int do_checkpoint(struct f2fs_sb_info *sbi, struct cp_control *cpc)
  {
  	struct f2fs_checkpoint *ckpt = F2FS_CKPT(sbi);